Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/EndOfYearDataManager.swift

```diff
@@ -0,0 +1,29 @@
+import FMDB
+import PocketCastsUtils
+
+/// Calculates user End of Year stats
+class EndOfYearDataManager {
+    private let endPeriod = "2022-12-01"
+
+    /// Returns the aproximately listening time for the current year
+    func listeningTime(dbQueue: FMDatabaseQueue) -> Double? {
+        var listeningTime: Double?
+
+        dbQueue.inDatabase { db in
+            do {
+                let query = "SELECT SUM(playedUpTo) as totalPlayedTime from \(DataManager.episodeTableName) WHERE lastPlaybackInteractionDate IS NOT NULL AND lastPlaybackInteractionDate BETWEEN strftime('%s', '2022-01-01') and strftime('%s', '\(endPeriod)')"
+                let resultSet = try db.executeQuery(query, values: nil)
+                defer { resultSet.close() }
+
+                while resultSet.next() {
```

Comment:
Nit: Since we're only expecting 1 row we don't need to use a while, we can just do an if.

```suggestion
                if resultSet.next() {
```